dockstore-webservice/src/main/java/io/dockstore/webservice/resources/OrganisationResource.java

```diff
@@ -102,13 +102,75 @@ public Organisation approveOrganisation(@ApiParam(hidden = true) @Auth User user
         return organisationDAO.findById(id);
     }
 
+    @GET
+    @Timed
+    @UnitOfWork
+    @Path("/name/{name}/")
+    @ApiOperation(value = "Retrieves an organisation by name.", notes = OPTIONAL_AUTH_MESSAGE, authorizations = { @Authorization(value = JWT_SECURITY_DEFINITION_NAME) }, response = Organisation.class)
+    public Organisation getOrganisationByName(@ApiParam(hidden = true) @Auth Optional<User> user,
+            @ApiParam(value = "Organisation name.", required = true) @PathParam("name") String name) {
+        if (!user.isPresent()) {
+            // No user given, only show approved organisations
+            Organisation organisation = organisationDAO.findApprovedByName(name);
+            if (organisation == null) {
+                String msg = "Organisation not found";
+                LOG.info(msg);
+                throw new CustomWebApplicationException(msg, HttpStatus.SC_NOT_FOUND);
+            }
+            return organisation;
+        } else {
+            // User is given, check if organisation is either approved or the user has access
+            // Admins and curators should be able to see unapproved organisations
+            boolean doesOrgExist = doesOrganisationByNameExistToUser(name, user.get().getId()) || user.get().getIsAdmin() || user.get().isCurator();
```

Comment:
I'm probably prematurely optimizing here, but this code flow is doing two database accesses in the happy path, one inside `doesOrganisationByNameExistToUser` and then the one on line 131.

You could just do line 131 at the beginning of this else, then if they are an admin or curator, or if they belong to the org, which you know from your `getUserOrgRole` method to see if the user belongs to the org, you can return the organization.